app/src/main/java/com/android/bookswap/model/userViewModel.kt

```diff
@@ -0,0 +1,174 @@
+package com.android.bookswap.model
+
+import android.location.Address
+import androidx.lifecycle.ViewModel
+import com.android.bookswap.data.DataUser
+
+open class UserViewModel(var email: String) : ViewModel() {
+  var uid = "ERROR_UUID"
+  private var dataUser = DataUser(email = email)
+  private var isLoaded = false
+  // private val firebaseConnection: FirebaseConnection = FirebaseConnection.getInstance()
+
+  open fun getUser(force: Boolean = false): DataUser {
+    if (!isLoaded || force) {
+      fetchUser()
+    }
+    return dataUser
+  }
+
+  private fun fetchUser() {
+    /*
+    firebaseConnection.getUserUidByEmail(user.email).addOnSuccessListener {
+        if (it.documents.isNotEmpty()) {
+            this.uid = it.documents[0].id
```

Comment:
If you are going to comment so much stuff, I think it would be better to remove it, it makes reading anything a lot more complicated.

Reply:
I thought it was all finished, please ignore this